src/pages/home/report/ReportActionsView.js

```diff
@@ -256,6 +257,15 @@ class ReportActionsView extends React.Component {
         return Visibility.isVisible() && !isSidebarCoveringReportView;
     }
 
+    // If the chat is optimistic (AKA not yet created) we don't need to call openChat again
+    openReportIfNecessary() {
+        if (this.props.report.isOptimisticReport) {
+            return;
+        }
+
+        Report.openReport(this.props.report.reportID);
+    }
```

Comment:
Can we add this check in Report.openReport? and then we can just call Report.openReport here

Reply:
Agree +1

Reply:
I thought about that, but `openReport` doesn't have the report object, so it doesn't know whether the report is optimistic or not.

Reply:
I still think we should do this check in `openReport`. Maybe we can pass an additional param `isOptimistic` to `openReport` in this case.

Reply:
So, If we add a parameter here, we end up calling:
```
Report.openReport(this.props.report.reportID, this.props.report.isOptimisticReport);
```

Instead of `openReportIfNecessary()`.

We could, instead, do:
``
this.props.report.isOptimisticReport || Report.openReport(this.props.report.reportID);
```

Which works out to the same amount of code, or we could do:
```
if (!this.props.report.isOptimisticReport) {
    Report.openReport(this.props.report.reportID);
}
```

Which is clearer but slightly longer, and seems to be the thing we're trying to avoid with this function.

However, adding a parameter to `openReport` to not open optimistic reports doesn't really make sense, we *do* want to open optimistic reports here:
https://github.com/Expensify/App/pull/11439/files#diff-f7eec01ae30248000c28c7d3f68c7ea00ba10e5fc64f2854474a350beb2e9556R147

It doesn't make sense to add an `isOptimistic` flag to `openReport` and then pass the wrong value at creation of the optimistic report.

I actually think the solution we have now is best, because it lets us explain why this is required in jsut one place, the comment above this function.